src/sentry/api/event_search.py

```diff
@@ -218,16 +220,16 @@ def visit_raw_search(self, node, children):
 
     def visit_numeric_filter(self, node, children):
         search_key, _, operator, search_value = children
-        # Tags are always strings
-        if not search_key.is_tag:
+        operator = operator[0] if not isinstance(operator, Node) else '='
+
+        if search_key.name in self.numeric_keys:
             try:
                 search_value = int(search_value.text)
             except ValueError:
                 raise InvalidSearchQuery('Invalid numeric query: %s' % (search_key,))
         else:
-            search_value = search_value.text
-
-        operator = operator[0] if not isinstance(operator, Node) else '='
+            search_value = operator + search_value.text if operator != '=' else search_value.text
```

Comment:
kind of wondering if we should raise here if `operator != '='` because it seems likely that they were maybe trying to use an invalid comparator for the column? don't need to handle now, just a thought

Reply:
Yeah, we might need to handle this on the basic search. Maybe we need to only accept operators as actual values inside of quotes.